bin/oneoff/backfill_data/user_cap_state_columns

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env ruby
+
+# This script migrates CAP data from properties to separate indexed columns.
+
+require 'ruby-progressbar'
+require_relative '../../../dashboard/config/environment'
+
+max_id = User.maximum(:id)
+batch_size = 100_000
+total_updated = 0
+
+progress_bar_title = proc {|num_updates = 0| "Processed[%c/%C]: |%W| Updated: #{total_updated += num_updates} %a"}
+progress_bar = ProgressBar.create(total: max_id, format: progress_bar_title.call)
+
+(1..max_id).step(batch_size) do |start_id|
+  finish_id = [start_id + batch_size.pred, max_id].min
+
+  User.select(:id).where(cap_state: nil).where("properties->>'$.child_account_compliance_state' IS NOT NULL").find_in_batches(start: start_id, finish: finish_id) do |users_batch|
```

Comment:
There are >100M rows in this table and likely only the most recent rows have a value in the `child_account_compliance_state` property). Can we safely assume that only students who have signed in within the last 2 years will have a value set? `last_sign_in_at` is an indexed column. Maybe we can do something like this:

``` ruby
User.
  select(:id).
  where('last_sign_in_at > ?', cutoff_date).
  where("properties->>'$.child_account_compliance_state' IS NOT NULL").
  in_batches(of: 1000) do |relation|
    relation.update_all <<~SQL.strip
       cap_state = IF(properties->>'$.child_account_compliance_state' = 's', 'l', properties->>'$.child_account_compliance_state'),
       cap_state_date = CAST(properties->>'$.child_account_compliance_state_last_updated' AS DATETIME)
     SQL
  end

Reply:
We should provision an adhoc environment and configure it to use a clone of the production database to test this backfill script.

Reply:
Any Colorado student under 13 is covered by the Colorado Privacy Act policy, regardless of when they were created.

Reply:
> Any Colorado student under 13 is covered by the Colorado Privacy Act policy, regardless of when they were created.

Are there Colorado students where the `child_account_compliance_state` property is NOT NULL that have not signed in over the last 2 years? Did we fill in that value for all Colorado students regardless of whether they have signed in?

Reply:
It's a great idea, Suresh! Thank you!

Reply:
Let's also `rescue` each attempt to `update_all` a batch, so that we can log it and move on to the next batch even if one set of updates fails.

Reply:
https://github.com/code-dot-org/code-dot-org/pull/59962/commits/ba87048a1cb6fd5835dea764fbe9dfac8d5617ca